lib/ansible/modules/cloud/azure/azure_rm_keyvault_facts.py

```diff
@@ -0,0 +1,193 @@
+        tags:
+          - testing
+'''
+
+RETURN = '''
+azure_keyvault:
+    description: List of key vault dicts.
+    returned: always
+    type: list
+    example:  []
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.common import AzureMissingResourceHttpError, AzureHttpError
+    from azure.mgmt.keyvault.models import ResourcePaged
+except:
+    # This is handled in azure_rm_common
+    pass
+
+AZURE_OBJECT_CLASS = 'KeyVault'
+
+
+class AzureRMKeyVaultFacts(AzureRMModuleBase):
+
+    def __init__(self):
+
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            name=dict(type='str'),
+            resource_group=dict(type='str'),
+            tags=dict(type='list'),
+            top=dict(type='int')
+        )
+
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_keyvault=[])
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+        self.top = None
+
+        super(AzureRMKeyVaultFacts, self).__init__(self.module_arg_spec)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if self.name and not self.resource_group:
+            self.fail("Parameter error: resource group required when filtering by name.")
+
+        # list the conditions and what to return based on user input
+
+        if self.name is not None:
+            # specific key vault
+            self.results['ansible_facts']['azure_keyvault'] = self.get_item()
+        elif self.resource_group:
+            # all the key vaults listed in that specific resource group
+            self.results['ansible_facts']['azure_keyvault'] = self.list_resource_group()
+        else:
+            # all the key vaults in a subscription
+            self.results['ansible_facts']['azure_keyvault'] = self.list_items()
+
+        return self.results
+
+    def get_item(self):
+        self.log('Get properties for {0}'.format(self.name))
+        item = None
+        results = []
+        # get specific key vault
+        try:
+            item = self.keyvault_client.vaults.get(self.resource_group, self.name)
+        except CloudError:
+            pass
+
+        # serialize result
+        if item and self.has_tags(item.tags, self.tags):
+            results = [item.as_dict()]
+        return results
+
+    def list_resource_group(self):
+        self.log('List items for resource group')
+        try:
+            response = self.keyvault_client.vaults.list_by_resource_group(self.resource_group, top=self.top)
+        except AzureHttpError as exc:
+            self.fail("Failed to list for resource group {0} - {1}".format(self.resource_group, str(exc)))
+
+        results = []
+        for item in response:
```

Comment:
use filter() for list

Reply:
@yuwzho can you give me an example?

Reply:
list comprehension: `my_list = [x for x in my_list if x.attribute == value]`
filter: `my_list = filter(lambda x: x.attribute == value, my_list)`

Reply:
Ah I follow know... thanks.